### PR TITLE
Remove `ObservableSet.linkedHashSetFrom()` and  use `Set` by default

### LINKED ISSUE #868: Use LinkedHashSet for ObservableSet default implementation.
### Discussed in https://github.com/mobxjs/mobx.dart/discussions/867

<div type='discussions-op-text'>

<sup>Originally posted by **pr-1** September 16, 2022</sup>
Flutter sky_engine by default uses LinkedHashSet for Set implementation. 
[https://api.flutter.dev/flutter/dart-collection/LinkedHashSet-class.html](url)

Why don't we use LinkedHashSet as default for ObservableSet constructors ?
[https://github.com/mobxjs/mobx.dart/blob/master/mobx/lib/src/api/observable_collections/observable_set.dart](url)

This causes confusion in analogy of Set and ObservableSet. Ideally they should behave same with just an observable wrapper.
But if you use the default ObservableSet it will not retain the order of iteration. on `addAll()` or any other method. 
Also I would add these things to the docs if we don't change  this behaviour.

I am aware of the alternate `ObservableSet.linkedHashSetFrom` provided separately in the api but ideally this should be the default one in my opinion. 

</div>

### PR DESCRIPTION
**Pull request by pr-1**
**`Fixes #868 `**.

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR
- [x] Run the **`set:versions` command** using `npm` or `yarn`. You can find this command in the `package.json` file in the root directory
- [x] Include the **necessary reviewers** for the PR

### CODE REVIEW

**Comment by pr-1** [comment]
@fzyzcjy  @pavanpodila Please have a look

File: mobx/lib/src/api/observable_collections/observable_set.dart
Comment (fzyzcjy) [inline mobx/lib/src/api/observable_collections/observable_set.dart:25]: Why "identity" here? Can we use `Set()`?

**Review by fzyzcjy — COMMENTED** [review]
LGTM % nits

File: mobx/lib/src/api/observable_collections/observable_set.dart
Comment (pr-1) [reply mobx/lib/src/api/observable_collections/observable_set.dart:25]: Yep `Set()` = `LinkedHashSet.identity()` so we can use any of these.

Comment (fzyzcjy) [reply mobx/lib/src/api/observable_collections/observable_set.dart:25]: Well, seems no. `Set() = LinkedHashSet()`. `Set.identity() = LinkedHashSet.identity()`.

Comment (pr-1) [reply mobx/lib/src/api/observable_collections/observable_set.dart:25]: Yes you're right I mis understood the identical comparer. Will push a hotfix, should be `Set()` here.
Thanks for that.

Comment (fzyzcjy) [reply mobx/lib/src/api/observable_collections/observable_set.dart:25]: You are welcome :)

File: mobx/lib/src/api/observable_collections/observable_set.dart
Comment (fzyzcjy) [inline mobx/lib/src/api/observable_collections/observable_set.dart:28]: LinkedHashSet -> Set

**Review by fzyzcjy — APPROVED** [review]

**Comment by fzyzcjy** [comment]
Approve after the final change

**Review by pavanpodila — CHANGES_REQUESTED** [review]
Need to run set:versions to set the version in the library. Also some conflicts need to be resolved

**Comment by pavanpodila** [comment]
resolve conflicts pls before merge @pr-1

**Comment by pr-1** [comment]
Updated version to 2.0.8 @pavanpodila  Please let me know if that is correct or this change should go in patch-count . Resolved conflicts.

File: mobx/CHANGELOG.md
Comment (pavanpodila) [inline mobx/CHANGELOG.md:2]: This is a big change. I suggest going to `2.1.0`

File: mobx/CHANGELOG.md
Comment (pavanpodila) [inline mobx/CHANGELOG.md:3]: LinkedHashSet must be in monocode. Its a class.

**Review by pavanpodila — CHANGES_REQUESTED** [review]

**Comment by pavanpodila** [comment]
Also there are some hygiene changes being pushed into main. Please take a pull from that as well (possibly in 30 mins). @pr-1

File: mobx/CHANGELOG.md
Comment (pavanpodila) [inline mobx/CHANGELOG.md:2]: can remove this word. Not adding much value here.

File: mobx/lib/mobx.dart
Comment (pavanpodila) [inline mobx/lib/mobx.dart:69]: this has been removed and moved to a separate `version.dart` file. Please remove this and move to that. You can run `melos run set_version` to sync up with `pubspec.yaml`

**Review by pavanpodila — CHANGES_REQUESTED** [review]

**Review by pavanpodila — APPROVED** [review]

**Pull request merged**
